tests/client/disco/TestAddonManager.js

```diff
@@ -106,4 +115,39 @@ describe('addonManager', () => {
       return addonManager.uninstall('test-id', {_mozAddonManager: fakeMozAddonManager});
     });
   });
+
+  describe('addChangeListener', () => {
+    const fakeEventCallback = sinon.stub();
+    fakeMozAddonManager = {
+      addEventListener: sinon.stub(),
+    };
+
+    const handleChangeEvent =
+      addonManager.addChangeListeners(fakeEventCallback, fakeMozAddonManager);
+    const eventMap = {
+      onDisabled: ON_DISABLE,
+      onEnabled: ON_ENABLE,
+      onInstalling: ON_INSTALLING,
+      onInstalled: ON_INSTALLED,
+      onUninstalling: ON_UNINSTALLING,
+      onUninstalled: ON_UNINSTALLED,
+    };
+
+    Object.keys(eventMap).forEach((event) => {
+      const action = eventMap[event];
+      it(`dispatches ${action}`, () => {
```

Comment:
dispatch might not be the best word to use for this. I think the callback actually calls dispatch?